crates/amaru-mempool/src/strategies/in_memory_mempool.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use amaru_kernel::cbor::Encode;
+use amaru_kernel::to_cbor;
+use amaru_ouroboros_traits::mempool::Mempool;
+use amaru_ouroboros_traits::{
+    CanValidateTransactions, MempoolSeqNo, TransactionValidationError, TxId, TxOrigin,
+    TxRejectReason, TxSubmissionMempool,
+};
+use std::collections::{BTreeMap, BTreeSet};
+use std::mem;
+use std::pin::Pin;
+use std::sync::Arc;
+use tokio::sync::Notify;
+
+/// A temporary in-memory mempool implementation to support the transaction submission protocol.
+///
+/// It stores transactions in memory, indexed by their TxId and by a sequence number assigned
+/// at insertion time.
+///
+/// The validation of the transactions are delegated to a `CanValidateTransactions` implementation.
+///
+pub struct InMemoryMempool<Tx> {
+    config: MempoolConfig,
+    inner: parking_lot::RwLock<MempoolInner<Tx>>,
+    notify: Notify,
+    tx_validator: Arc<dyn CanValidateTransactions<Tx>>,
+}
+
+impl<Tx> Default for InMemoryMempool<Tx> {
+    fn default() -> Self {
+        InMemoryMempool {
+            config: MempoolConfig::default(),
+            inner: parking_lot::RwLock::new(MempoolInner::default()),
+            notify: Notify::new(),
+            tx_validator: Arc::new(DefaultCanValidateTransactions),
+        }
+    }
+}
+
+impl<Tx> InMemoryMempool<Tx> {
+    pub fn new(config: MempoolConfig, tx_validator: Arc<dyn CanValidateTransactions<Tx>>) -> Self {
+        InMemoryMempool {
+            config,
+            inner: parking_lot::RwLock::new(MempoolInner::default()),
+            notify: Notify::new(),
+            tx_validator,
+        }
+    }
+
+    pub fn from_config(config: MempoolConfig) -> Self {
+        Self::new(config, Arc::new(DefaultCanValidateTransactions))
+    }
+}
+
+/// A default transactions validator.
+#[derive(Clone, Debug, Default)]
+pub struct DefaultCanValidateTransactions;
```

Comment:
This highly dangerous code should be named more scarily, e.g. `AcceptAnyTransactionWhatsoever`.